src/instructlab/sdg/llmblock.py

```diff
@@ -113,21 +132,30 @@ def generate(self, samples, **gen_kwargs) -> Dataset:
         # validate each sample
         for sample in samples:
             if not self._validate(self.prompt_template, sample):
-                return None
+                logger.warning("Sample failed validation")  #TODO add details
+                #TODO remove sample from samples
```

Comment:
Hmm. Should this be in a separate PR. If in this PR, the TODO should be resolved?